bredis/AsyncConnection.hpp

```diff
@@ -51,24 +53,75 @@ template <typename S> class AsyncConnection {
     std::mutex tx_queue_mutex_;
     std::mutex rx_queue_mutex_;
 
+    // I think that queuing should be the responsibility
+    // of a higher level interface and not this class.
+    // Instead, this class should just allow for one
+    // active asynchronous Redis read and one active
+    // asynchronous Redis write, with undefined behavior
+    // if two concurrent reads or two concurrent writes
+    // happen.
+    //
+    // Rationale: TCP/IP provides application-level flow
+    // control to match the rate of the receiver and the
+    // sender. In this implementation, the reader attempts
+    // to read as quickly as possible no matter how fast
+    // the consumer (in this case the command_callback_t)
+    // can process the data. This not only circumvents
+    // TCP/IP's natural application level flow control,
+    // but allows the size of the respective queues to
+    // grow without bounds. Furthermore, the caller is
+    // given no opportunity to provide "back pressure",
+    // nor is the caller able to detect the unbounded
+    // growth and impose its own policies on it.
+    // 
```

Comment:
@vinniefalco The idea is cool, i.e. allows to get more fine-gained control. 

I have tried to implement it on separate branch `try-avoid-queuing`, namely:

- https://github.com/basiliscos/cpp-bredis/blob/try-avoid-queuing/include/bredis/AsyncConnection.hpp
- https://github.com/basiliscos/cpp-bredis/blob/try-avoid-queuing/include/bredis/impl/async_connection.ipp
- https://github.com/basiliscos/cpp-bredis/blob/try-avoid-queuing/t/11-multi-ping.cpp

The resulting performance for 100k pings dropped from `0.27s` to `2.04s`. The rationale behind that is that it half-switched from pipe-lining

https://redis.io/topics/pipelining

i.e. for each "async_write"  with 6-7 bytes of `ping` command it performs `write` syscall (or something like). Meanwile in the original implementation the `tx_buffer` grew "in infinitum", what allowed to do `write` syscall much less frequently.

Trying to find compromise...